toolchain/lex/lex.cpp

```diff
@@ -1179,17 +1175,43 @@ auto Lexer::LexWordAsTypeLiteralToken(llvm::StringRef word, int32_t byte_offset)
       return LexResult::NoMatch();
   };
 
+  // No leading zeros allowed.
+  if ('1' > word[1] || word[1] > '9') {
+    return LexResult::NoMatch();
+  }
+
   llvm::StringRef suffix = word.substr(1);
-  if (!CanLexInt(emitter_, suffix)) {
+  constexpr ssize_t DigitLimit = std::numeric_limits<int64_t>::digits10;
```

Comment:
You use `int64_t` here but `suffix_value` is `ssize_t`. Should the same type be used?

Reply:
Yeah, that was an accident. I've restructured the code so it constructively uses the right type.